trieste/bayesian_optimizer.py

```diff
@@ -667,14 +667,25 @@ def optimize(
                             "will be available."
                         ) from e
 
-                with Timer() as total_step_wallclock_timer:
+                if step == 1 and fit_initial_model:
                     with Timer() as initial_model_fitting_timer:
-                        if step == 0 and fit_initial_model:
+                        for tag, model in models.items():
+                            dataset = datasets[tag]
+                            model.update(dataset)
+                            model.optimize(dataset)
+                    if summary_writer:
+                        logging.set_step_number(0)
+                        with summary_writer.as_default(step=0):
                             for tag, model in models.items():
-                                dataset = datasets[tag]
-                                model.update(dataset)
-                                model.optimize(dataset)
+                                with tf.name_scope(f"{tag}.model"):
```

Comment:
maybe move this to a `_write_summary_initial_fit` function (to go alongside `_write_summary_step` and `_write_summary_init`)?

Reply:
done